Userland/Applications/Piano/KnobsWidget.cpp

```diff
@@ -140,18 +139,14 @@ KnobsWidget::KnobsWidget(TrackManager& track_manager, MainWidget& main_widget)
         m_release_value->set_text(String::number(new_release));
     };
 
-    m_delay_knob = m_knobs_container->add<GUI::VerticalSlider>();
-    m_delay_knob->set_tooltip("Delay speed, 0 = off");
-    m_delay_knob->set_range(0, delay_max);
-    m_delay_knob->set_value(delay_max - m_track_manager.current_track().delay());
-    m_release_knob->set_step(1);
-    m_delay_knob->on_change = [this](int value) {
-        int new_delay = delay_max - value;
-        if (m_change_underlying)
-            m_track_manager.current_track().set_delay(new_delay);
-        VERIFY(new_delay == m_track_manager.current_track().delay());
-        m_delay_value->set_text(String::number(new_delay));
-    };
+    for (auto& raw_parameter : m_track_manager.current_track().delay()->parameters()) {
+        // FIXME: We shouldn't do that, but we know the effect and it's nice.
+        auto& parameter = static_cast<LibDSP::ProcessorRangeParameter&>(raw_parameter);
+        m_delay_values.append(m_values_container->add<GUI::Label>(String::number(static_cast<double>(parameter.value()))));
+        auto& parameter_knob_value = m_delay_values.last();
+        m_delay_labels.append(m_labels_container->add<GUI::Label>(String::formatted("Delay: {}", parameter.name())));
```

Comment:
This does not show up correctly in the UI (it shows up as `Delay...` for all three parameters).
Perhaps a newline between the parameter name and "Delay" could fix that?